### ISSUE TITLE
unused -mf16c compiler flag during android build

### ISSUE THREAD

**Issue by larroy** [apache/incubator-mxnet#15935]
```
ci/build.py -p android_armv7

[...]

[98/290] ccache /usr/arm-linux-androideabi/bin/arm-linux-androideabi-clang++  -DDMLC_USE_CXX11 -DDMLC_USE_CXX11=1 -DMSHADOW_INT64_TENSOR_SIZE=0 -DMSHADOW_IN_CXX11 -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_CUDA=0 -DMSHADOW_USE_MKL=0 -DMSHADOW_USE_SSE=0 -DMXNET_USE_BLAS_OPEN=1 -DMXNET_USE_NCCL=0 -DMXNET_USE_OPENCV=0 -DMXNET_USE_SIGNAL_HANDLER=1 -DNDEBUG=1 -I/work/mxnet/include -I/work/mxnet/src -isystem /usr/arm-linux-androideabi/include -I/work/mxnet/3rdparty/mshadow -I/work/mxnet/3rdparty/nvidia_cub -I/work/mxnet/3rdparty/tvm/nnvm/include -I/work/mxnet/3rdparty/tvm/include -I/work/mxnet/3rdparty/dmlc-core/include -I/work/mxnet/3rdparty/dlpack/include -D_GLIBCXX_ASSERTIONS  -mf16c -Wall -Wno-unknown-pragmas -Wno-sign-compare -Wno-braced-scalar-init -O3 -g -std=c++11 -mf16c -std=c++0x -O2 -g -DNDEBUG -fPIC -MD -MT CMakeFiles/mxnet_static.dir/src/c_api/c_api_function.cc.o -MF CMakeFiles/mxnet_static.dir/src/c_api/c_api_function.cc.o.d -o CMakeFiles/mxnet_static.dir/src/c_api/c_api_function.cc.o -c /work/mxnet/src/c_api/c_api_function.cc
clang60++: warning: argument unused during compilation: '-mf16c' [-Wunused-command-line-argument]
clang60++: warning: argument unused during compilation: '-mf16c' [-Wunused-command-line-argument]
```

**Comment by larroy** [comment]
@mxnet-label-bot add [Android, Build,Good First Issue]

**Comment by larroy** [comment]
@mxnet-label-bot add [Call for Contribution]

**Comment by michiboo** [comment]
Hi is this issue still available?

**Comment by larroy** [comment]
@michiboo did you try to rebuild? ci/build.py -p android_armv7

**Comment by larroy** [comment]
Still happening @michiboo would you like to help?

**Comment by michiboo** [comment]
@larroy Hi Yes I would like to still work on it

**Comment by larroy** [comment]
@michiboo thank you. Could you modify CMakeFile to omit these flags for the architectures where it doesn't apply?

**Comment by michiboo** [comment]
@larroy  Hi for some reason, I cannot reproduce the warning in this build

**Comment by larroy** [comment]
What did you try?  ci/build.py -p android_armv7  should reproduce.

**Comment by michiboo** [comment]
I try the exact same command, but I ended up with a error. I checked I already has opencv 4 so it shouldn't be a problem

+ cmake -DANDROID=ON -DUSE_CUDA=OFF -DUSE_SSE=OFF -DSUPPORT_F16C=OFF -DUSE_LAPACK=OFF -DUSE_OPENCV=OFF -DUSE_OPENMP=OFF -DUSE_SIGNAL_HANDLER=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DUSE_MKL_IF_AVAILABLE=OFF -G Ninja /work/mxnet
-- CMAKE_CROSSCOMPILING FALSE
-- CMAKE_HOST_SYSTEM_PROCESSOR x86_64
-- CMAKE_SYSTEM_PROCESSOR x86_64
-- CMAKE_SYSTEM_NAME Linux
-- CMake version '3.13.2' using generator 'Ninja'
-- Determining F16C support
-- F16C enabled
-- Found OpenBLAS libraries: /usr/arm-linux-androideabi/lib/libopenblas.a
-- Found OpenBLAS include: /usr/arm-linux-androideabi/include
-- OpenCV Disabled
CMake Error at CMakeLists.txt:491 (add_subdirectory):
  add_subdirectory given source "/work/mxnet/3rdparty/googletest/googletest"
  which is not an existing directory.


CMake Warning at CMakeLists.txt:847 (message):
  OpenCV_VERSION_MAJOR: , version 3 with imgcodecs is required for im2rec,
  im2rec will not be available


CMake Warning at example/image-classification/predict-cpp/CMakeLists.txt:3 (message):
  OpenCV version >= 3 should be enabled and found to build image
  classification example, skipping...


-- Configuring incomplete, errors occurred!
See also "/work/build/CMakeFiles/CMakeOutput.log".
2020-02-04 15:57:17,189 - root - INFO - Waiting for status of container d73f47c477fe for 600 s.
2020-02-04 15:57:17,303 - root - INFO - Container exit status: {'Error': None, 'StatusCode': 1}
2020-02-04 15:57:17,303 - root - ERROR - Container exited with an error 😞
2020-02-04 15:57:17,303 - root - INFO - Executed command for reproduction:

./ci/build.py -p android_armv7

2020-02-04 15:57:17,304 - root - INFO - Stopping container: d73f47c477fe
2020-02-04 15:57:17,305 - root - INFO - Removing container: d73f47c477fe
2020-02-04 15:57:17,322 - root - CRITICAL - Execution of ['/work/mxnet/ci/docker/runtime_functions.sh', 'build_android_armv7'] failed with status: 1
